indexer/src/transform.test.ts

```diff
@@ -0,0 +1,606 @@
+  assertEquals,
+  assertExists,
+} from "https://deno.land/std@0.213.0/assert/mod.ts";
+import transform from "./main.ts";
+import { toEthHeader } from "./types/header.ts";
+import { Bloom } from "./deps.ts";
+import { padString, toHexString } from "./utils/hex.ts";
+import {
+  toTypedEthTx,
+  typedTransactionToEthTx,
+} from "../src/types/transaction.ts";
+import { JsonRpcLog, toEthLog } from "./types/log.ts";
+import { toEthReceipt } from "./types/receipt.ts";
+import {
+  BlockHeader,
+  Event,
+  EventWithTransaction,
+  JsonRpcTx,
+  Transaction,
+  TransactionReceipt,
+  TransactionWithReceipt,
+  TypedTransaction,
+} from "./deps.ts";
+
+const jsonData = await Deno.readTextFile(
+  "indexer/src/test-data/transactionsData.json",
+);
+const transactionsData = JSON.parse(jsonData);
+
+async function transformData(
+  header: BlockHeader,
+  events: EventWithTransaction[],
+  transactions: TransactionWithReceipt[],
+  transaction: Transaction,
+  receipt: TransactionReceipt,
+  event: Event,
+) {
+  const result = await transform({ header, events, transactions });
+
+  const expectedHeader = await toEthHeader({
+    header: header,
+    gasUsed: BigInt(0),
+    logsBloom: new Bloom(new Uint8Array(256)),
+    receiptRoot: new Uint8Array(),
+    transactionRoot: new Uint8Array(),
+    blockNumber: padString(toHexString(header.blockNumber), 8),
+    blockHash: header.blockHash,
+    isPendingBlock: false,
+  });
+  const typedEthTx = toTypedEthTx({ transaction });
+  const ethTx = typedTransactionToEthTx({
+    typedTransaction: typedEthTx as TypedTransaction,
+    receipt,
+    blockNumber: padString(toHexString(header.blockNumber), 8),
+    blockHash: header.blockHash,
+    isPendingBlock: false,
+  });
+
+  const ethLogs = receipt.events
+    .map((e: Event) => {
+      return toEthLog({
+        transaction: ethTx as JsonRpcTx,
+        event: e,
+        blockNumber: padString(toHexString(header.blockNumber), 8),
+        blockHash: header.blockHash,
+        isPendingBlock: false,
+      });
+    })
+    .filter((e: JsonRpcLog | null) => e !== null) as JsonRpcLog[];
+
+  const ethLogsIndexed = ethLogs.map((log, index) => {
+    log.logIndex = index.toString();
+    return log;
+  });
+
+  const ethReceipt = toEthReceipt({
+    transaction: ethTx as JsonRpcTx,
+    logs: ethLogsIndexed,
+    event,
+    cumulativeGasUsed: 0n,
+    blockNumber: padString(toHexString(header.blockNumber), 8),
+    blockHash: header.blockHash,
+    isPendingBlock: false,
+  });
+
+  return { result, expectedHeader, ethTx, ethLogs, ethReceipt };
+}
+const header: BlockHeader = {
+  blockNumber: "2000",
+  blockHash:
+    "0x0286731b9083ab0be4a875726f3da0c3c6be2a909f980bd6dccdabe75dde18f5",
+  parentBlockHash:
+    "0x729fefc61158e655d04e50ca6d1df25970a258cf3a88cfad5579a44907d52a2",
+  newRoot: "0x20882c140e56616f37764e40afe2961894e6146777f0a01a565ac06bb04b703",
+  timestamp: "1717093503",
+  sequencerAddress: "0x1",
+};
+const transaction: Transaction = {
```

Comment:
maybe we should name this `mock_transaction` wdyt?

Reply:
renamed all the variables to: mockHeader, mockTransaction, mockReceipt, mockEvent, mockEvents, mockTransactions
thanks